src/wp-includes/class-wp-block.php

```diff
@@ -310,6 +311,15 @@ private function process_block_bindings() {
 			}
 		}
 
+		/*
+		 * Update the bindings metadata of the computed attributes.
+		 * This ensures the block receives the expanded __default binding metadata when it renders.
+		 */
+		$computed_attributes['metadata'] = array_merge(
+			$parsed_block['attrs']['metadata'],
+			array( 'bindings' => $bindings )
+		);
+
```

Comment:
Should we move this code to [the conditional that handles the __default](https://github.com/WordPress/wordpress-develop/blob/cd2f3f06daeaaefcfb646196f448e1d25a39b7c5/src/wp-includes/class-wp-block.php#L266-L288)?

If I am not mistaken, it should work, and it could help clarify that it is related to that specific use case.

Reply:
Makes sense. I've pushed a commit with that change.